### PR TITLE
Add  non-square kernel support to median filter

### PR DESCRIPTION
**Pull request by TH3CHARLie**
I add non-square kernel support for the median filter. It now supports any rectangle shape kernels with odd side lengths.

### CODE REVIEW

File: src/filter/median.rs [hunk 1]
```diff
@@ -2,10 +2,10 @@ use image::{GenericImageView, Pixel};
 use crate::definitions::Image;
 use std::cmp::{min, max};
 
-/// Applies a median filter of given `radius` to an image. Each output pixel is the median
-/// of the pixels in a `2 * radius + 1` square of pixels in the input image.
+/// Applies a median filter of given kernel to an image. Each output pixel is the median
```
Comment (theotherphil) [inline src/filter/median.rs]: "kernel" is wrong here. "of given dimensions" would be ok.

File: src/filter/median.rs [hunk 2]
```diff
@@ -72,10 +72,41 @@ use std::cmp::{min, max};
...
-pub fn median_filter<P>(image: &Image<P>, radius: u32) -> Image<P>
+/// 
+/// ```
+/// # extern crate image;
+/// # #[macro_use]
+/// # extern crate imageproc;
+/// # fn main() {
+/// use imageproc::filter::median_filter;
+///
+/// // The kernel can be non-square shape.
```
Comment (theotherphil) [inline src/filter/median.rs]: This line is not necessary

File: src/filter/median.rs [hunk 3]
```diff
@@ -289,43 +322,50 @@ mod tests {
...
+    bench_median_filter!(bench_median_filter_s100_r8, side: 100, x_radius: 8,y_radius: 8);
+    
+    // benchmark on non-square kernels
+    bench_median_filter!(bench_median_filter_s100_rx1_ry4, side: 100, x_radius: 1,y_radius: 4);
+    bench_median_filter!(bench_median_filter_s100_rx1_ry8, side: 100, x_radius: 1,y_radius: 8);
+    bench_median_filter!(bench_median_filter_s100_rx4_ry8, side: 100, x_radius: 4,y_radius: 1);
+    bench_median_filter!(bench_median_filter_s100_rx8_ry1, side: 100, x_radius: 8,y_radius: 1);
+
 
-    // Reference implementation of median filter - written to be as simple as possible,
```
Comment (theotherphil) [inline src/filter/median.rs]: Why did you remove this comment?

File: src/filter/median.rs [hunk 4]
```diff
@@ -348,20 +388,21 @@ mod tests {
     fn median(sorted: &[u8]) -> u8 {
         let mid = sorted.len() / 2;
         sorted[mid]
-    }
+    }
```
Comment (theotherphil) [inline src/filter/median.rs]: Bad whitespace

File: src/filter/median.rs (hunk 3 above)
Comment (TH3CHARLie) [reply src/filter/median.rs]: Unintentionally deleted it, recover it in the new commit
